pkg/agent/externalnode/external_node_controller_linux_test.go

```diff
@@ -0,0 +1,1155 @@
+
+func TestAddExternalNode(t *testing.T) {
+	defer mockGetIPNetDeviceFromIP([]mockGetIPNetDeviceFromIPParam{
+		{&net.Interface{Name: "if1"}, nil},
+	})()
+	c := newExternalNodeController(t)
+	externalNode := &v1alpha1.ExternalNode{
+		ObjectMeta: metav1.ObjectMeta{Name: "eeName", Namespace: "ns1", Labels: map[string]string{"en": "vm1"}},
+		Spec: v1alpha1.ExternalNodeSpec{
+			Interfaces: []v1alpha1.NetworkInterface{{IPs: []string{"1.1.1.2"}}},
+		},
+	}
+	mockIfaceStore.EXPECT().GetInterfaceByName("if1").Return(&if1, true)
+	mockOVSBridgeClient.EXPECT().GetPortData(portUUID1, "if1").Return(portData1, nil)
+	expectedInfo := map[string]interface{}{
+		"uplink-name":      "uplinkName",
+		"entity-name":      "eeName",
+		"antrea-type":      "host",
+		"entity-namespace": "externalEntityNamespace",
+		"ip-address":       "1.1.1.2",
+		"uplink-port":      portUUID2,
+	}
+	mockOVSBridgeClient.EXPECT().SetPortExternalIDs(if1.InterfaceName, expectedInfo)
+	iface := &interfacestore.InterfaceConfig{
+		InterfaceName: if1.InterfaceName,
+		Type:          interfacestore.ExternalEntityInterface,
+		OVSPortConfig: &interfacestore.OVSPortConfig{
+			PortUUID: if1.PortUUID,
+			OFPort:   if1.OFPort,
+		},
+		EntityInterfaceConfig: &interfacestore.EntityInterfaceConfig{
+			EntityName:      "eeName",
+			EntityNamespace: if1.EntityNamespace,
+			UplinkPort: &interfacestore.OVSPortConfig{
+				PortUUID: if1.UplinkPort.PortUUID,
+				OFPort:   if1.UplinkPort.OFPort,
+			},
+		},
+		IPs: []net.IP{net.ParseIP("1.1.1.2")},
+	}
+	mockIfaceStore.EXPECT().AddInterface(iface)
+	c.addExternalNode(externalNode)
+	assert.Equal(t, externalNode, c.syncedExternalNode)
+}
+
+func TestEnqueueExternalNodeUpdate(t *testing.T) {
+	c := newExternalNodeController(t)
+	externalNode1 := &v1alpha1.ExternalNode{
+		ObjectMeta: metav1.ObjectMeta{Name: "vm1", Namespace: "ns1", Labels: map[string]string{"en": "vm1"}},
+		Spec: v1alpha1.ExternalNodeSpec{
+			Interfaces: []v1alpha1.NetworkInterface{{IPs: []string{"1.1.1.2"}}},
+		},
+	}
+	externalNode2 := &v1alpha1.ExternalNode{
+		ObjectMeta: metav1.ObjectMeta{Name: "vm1", Namespace: "ns1", Labels: map[string]string{"en": "vm1"}},
+		Spec: v1alpha1.ExternalNodeSpec{
+			Interfaces: []v1alpha1.NetworkInterface{{IPs: []string{"1.1.1.2"}}},
+		},
+	}
+	externalNode3 := &v1alpha1.ExternalNode{
+		ObjectMeta: metav1.ObjectMeta{Name: "vm1", Namespace: "ns1", Labels: map[string]string{"en": "vm1"}},
+		Spec: v1alpha1.ExternalNodeSpec{
+			Interfaces: []v1alpha1.NetworkInterface{{IPs: []string{"1.1.1.3"}}},
+		},
+	}
+	for _, tt := range []struct {
+		name           string
+		oldObj         *v1alpha1.ExternalNode
+		newObj         *v1alpha1.ExternalNode
+		expectEnqueued bool
+	}{
+		{
+			name:           "different externalnode",
+			newObj:         externalNode2,
+			oldObj:         externalNode1,
+			expectEnqueued: false,
+		},
+		{
+			name:           "same externalnode",
+			newObj:         externalNode3,
+			oldObj:         externalNode1,
```

Comment:
I suppose these two should be  `externalNode2` and `externalNode1`,  or you should exchange the test names.

Reply:
updated